### PR TITLE
JDG-2214 Custom User Configuration

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/JDG-2214

@galderz Now issued against the datagrid73-dev branch. The example template has been moved to a documentation subdirectory and converted to yaml.

### CODE REVIEW

File: docs/datagrid-user-xml.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+3. Expose the required services and ports in your template
+
+When `USER_CONFIG_MAP=true` our image copies the contents of `opt/datagrid/standalone/configuration/user`. i.e. the configmap, to `/opt/datagrid/standalone/configuration`.
+This means that in addition to passing `standalone.xml`, it is also possible to provide `logging.properties`, `application-role.properties` etc in the configmap.
+
+Creating a configmap containing the desired configuration files is trivial, as an entire directory can be used to create a configmap consisting of filenames and their contents as key/values in the configmap. This can be achieved via `oc create configmap <name-of-configmap> --from-file=<local-dir>`.
+
+To create an application that uses a custom configuration, login to your openshift cluster via `oc`, push your built image to the cluster's repository and execute the following:
+
+1. `oc create -f docs/examples/user-configuration/user-config-template.yaml`
```
Comment (galderz) [inline docs/datagrid-user-xml.adoc]: Is `user-config-template.yaml` right? Seems like the file is called `user-xml-template.yaml` ?

**Review by galderz — CHANGES_REQUESTED** [review]
Just one minor comment.

@ryanemerson Is @oraNod eventually putting this docu in the official docu? If so would we delete stuff from the repo? Or move it to a tutorial?

**Comment by oraNod** [comment]
@galderz @ryanemerson I've a task, JDG-2240, to cover the custom user config in the main OpenShift docs. I'd like to eventually remove the docs from this repo and point to the product docs.

**Comment by ryanemerson** [comment]
@galderz Good spot about the name, I had missed that as part of the refactor. The .adoc and template files are both `*user-config.*`.

@galderz @oraNod Regarding the docs in general, shall we merge this as is and then delete the docs with JDG-2240? That way Don has these as a basis at least until the tutorials and prod documents are in place.

**Comment by ryanemerson** [comment]
I also updated https://github.com/jboss-container-images/jboss-datagrid-7-openshift-image/pull/114

**Comment by galderz** [comment]
Looks good now, integrating.

**Review by galderz — APPROVED** [review]

**Pull request merged**